"precommit" on SQL security "with-ssl" and "without-ssl" projects fails after a "gradle clean"

To reproduce this:
1. `gradlew clean`
2. `gradlew :x-pack:plugin:sql:qa:security:without-ssl:precommit --parallel`
or
`gradlew :x-pack:plugin:sql:qa:security:with-ssl:precommit --parallel`

This will fail with 

```
> Task :x-pack:plugin:sql:qa:security:without-ssl:testingConventions FAILED
Expected at least one test class included in task :x-pack:plugin:sql:qa:security:without-ssl:integTestRunner, but found none.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':x-pack:plugin:sql:qa:security:without-ssl:testingConventions'.
> Testing conventions are not honored
```

Digging into this a bit, it seems the PR that fixed https://github.com/elastic/elasticsearch/issues/40550 seems to be the one that introduced the behavior above.

CC @mark-vieira @atorok

--- a/buildSrc/src/main/java/org/elasticsearch/gradle/precommit/TestingConventionsTasks.java
+++ b/buildSrc/src/main/java/org/elasticsearch/gradle/precommit/TestingConventionsTasks.java
@@ -65,7 +65,7 @@ public class TestingConventionsTasks extends DefaultTask {
     public TestingConventionsTasks() {
         setDescription("Tests various testing conventions");
         // Run only after everything is compiled
-        Boilerplate.getJavaSourceSets(getProject()).all(sourceSet -> dependsOn(sourceSet.getClassesTaskName()));
+        Boilerplate.getJavaSourceSets(getProject()).all(sourceSet -> dependsOn(sourceSet.getOutput().getClassesDirs()));
         naming = getProject().container(TestingConventionRule.class);
     }
 

--- a/x-pack/plugin/sql/qa/security/build.gradle
+++ b/x-pack/plugin/sql/qa/security/build.gradle
@@ -15,6 +15,7 @@ subprojects {
     test {
       mainProject.sourceSets.test.output.classesDirs.each { dir ->
         output.addClassesDir { dir }
+        output.builtBy(mainProject.tasks.testClasses)
       }
       runtimeClasspath += mainProject.sourceSets.test.output
     }

